turbinia/evidence.py

```diff
@@ -300,19 +341,37 @@ class GoogleCloudDiskRawEmbedded(GoogleCloudDisk):
 
   Attributes:
     embedded_path: The path of the raw disk image inside the Persistent Disk
+    _clouddisk_mount_path(str): the path where the attached disk is mounted.
   """
 
   def __init__(self, embedded_path=None, *args, **kwargs):
     """Initialization for Google Cloud Disk."""
     self.embedded_path = embedded_path
     super(GoogleCloudDiskRawEmbedded, self).__init__(*args, **kwargs)
 
+    self._clouddisk_mount_path = None
+
   def _preprocess(self):
-    self.local_path = google_cloud.PreprocessAttachDisk(self.disk_name)
-    self.loopdevice_path = mount_local.PreprocessLosetup(self.local_path)
-    self.mount_path = mount_local.PreprocessMountDisk(
+    self._attached_path = google_cloud.PreprocessAttachDisk(self.disk_name)
+    self.loopdevice_path = mount_local.PreprocessLosetup(self._attached_path)
+    self._clouddisk_mount_path = mount_local.PreprocessMountDisk(
         self.loopdevice_path, self.mount_partition)
-    self.local_path = os.path.join(self.mount_path, self.embedded_path)
+
+    # If the mounting of the embedded image step fails, at least local_path
+    # still points to the image.
+    self.local_path = os.path.join(
+        self._clouddisk_mount_path, self.embedded_path)
+
+    try:
+      # We now try to mount the image pointed by self.embedded_path
+      self.mount_path = mount_local.PreprocessMountDisk(
+          self.embedded_path, self.mount_partition)
+      self.local_path = self.mount_path
+      self.is_mounted = True
+    except TurbiniaException as e:
+      log.error(
+          'Could not mount partition {0:d} for GoogleCloudDiskRawEmbedded '
+          '"{1:s}": {2!s}'.format(self.mount_partition, self.disk_name, e))
```

Comment:
This error message is going to be somewhat confusing because the `mount_partition` is almost always going to be None in this preprocessor method (unless the user sets the flag), and it's only in the preprocessor itself that it will default to using `1`.  Maybe we should remove the partition here, and then make sure that the processors are including the full device path in the error message when they raise the exception?